src/sendReport.js

```diff
@@ -37,6 +36,11 @@ function sendReport(requestBody, config, ipAddress) {
       .on('error', err => {
         reject(err);
       });
+    if (Number.isInteger(config.networkTimeout) && config.networkTimeout > 0) {
```

Comment:
I believe this will effectively turn this block into a no-op when network timeout is set via environment variables because environment variables are always strings. Would instead check the config implementation and cast as an integer if possible or fallback to the default value.

Reply:
The default config returns `5000`, the environment does `parseInt()` then checks if it is an integer, Cosmi just checks if it is an integer. The test that fails just uses a JavaScript object and not any of the actual config classes, so in this case we could put in `parseInt()` here as well to wrap the timeouts.

Reply:
Ok, so how is the non-integer value getting there? Want to make sure we're not handling a case that shouldn't happen (non-integer network timeout).

Reply:
For the test that was failing with a TypeError (`report.test.js`) it just uses a plain object for the config and not one of the classes the normal agent uses. So since it is not specified there it would be `null`. Should that test then just specify some really high timeout value then?

Reply:
Object configs are a part of the config implementation: https://github.com/iopipe/iopipe-js-core/blob/master/src/config/object.js#L58

So casting that object value as an integer if possible, falling back to default value if not.

Reply:
So should this test https://github.com/iopipe/iopipe-js-core/blob/master/src/report.test.js#L11 be using the object configuration then? Note that this PR does not touch that test at all.

Reply:
I think instead of the condition to handle a non-integer network timeout, we should just handle that in the config implementation itself. It became a part of the scope of this PR when `.setTimeout()` started complaining about a non-integer value.